fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -195,4 +223,243 @@ public void getTriplesOfBinaryDescription() throws Exception {
         assertEquals(dcTitleTriple, retrievedUserStream.findFirst().get());
     }
 
+    @Test
+    public void persistFailsIfCommitAlreadyComplete() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn("ocfl-object-id");
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        //perform the create rdf operation
+        session.persist(rdfSourceOperation);
+
+        //commit to OCFL
+        session.commit(CommitOption.UNVERSIONED);
+
+        //this should fail
+        try {
+            session.persist(rdfSourceOperation);
+            fail("second session.persist(...) invocation should have failed.");
+        } catch (PersistentStorageException ex) {
+            //expected failure
+        }
+    }
+
+    @Test
+    public void getTriplesFailsIfAlreadyCommitted() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn("ocfl-object-id");
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        //perform the create rdf operation
+        session.persist(rdfSourceOperation);
+
+        //commit to OCFL
+        session.commit(CommitOption.UNVERSIONED);
+
+        //this should fail
+        try {
+            session.getTriples(resourceId, null);
+            fail("second session.getTriples(...) invocation should have failed.");
+        } catch (PersistentStorageException ex) {
+            //expected failure
+        }
+    }
+
+    @Test
+    public void commitFailsIfAlreadyCommitted() throws PersistentStorageException {
```

Comment:
I don't think its a problem, but Mohamed's PR takes the approach committing a committed transaction does nothing. Same with rolling back a rolled back transaction. Just mentioning that in case you'd prefer consistency, but its probably fine for the persistence layer to be less forgiving.

Reply:
We can revisit if it becomes a problem.  I think it's okay for now.